test/test_HeadingController.py

```diff
@@ -0,0 +1,354 @@
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+        # Adjust parameters
+        mock_time = 2 * sailbot_constants.SWITCH_INTERVAL
+
+        # Switch occurs
+        self.assertTrue(hc.switchControlMode(
+            heading_error=mock_heading_error,
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+    # TACKABLE tests
+
+    # Entry to TACKABLE from UNKNOWN
+
+    def test_initialize_tackableMode(self):
+        mock_speed = sailbot_constants.SPEED_THRESHOLD_FOR_JIBING_KNOTS + 0.1
+        mock_time = 0
+
+        hc = HeadingController(mock_speed, mock_time, ControlModes.UNKNOWN.value)
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+    # Remain in TACKABLE since exit condition is not met
+
+    def test_noExit_tackableMode(self):
+        mock_speed = sailbot_constants.SPEED_THRESHOLD_FOR_JIBING_KNOTS + 0.1
+        mock_time = 0
+        mock_heading_error = 2 * sailbot_constants.MIN_HEADING_ERROR_FOR_SWITCH
+
+        # Enter TACKABLE mode from UNKNOWN
+        hc = HeadingController(mock_speed, mock_time, ControlModes.UNKNOWN.value)
+
+        # No switch
+        self.assertFalse(hc.switchControlMode(
+            heading_error=mock_heading_error,
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+        # Adjust parameters
+        mock_speed *= 2
+        mock_time += 0.5 * sailbot_constants.SWITCH_INTERVAL
+        mock_heading_error = 1.5 * sailbot_constants.MIN_HEADING_ERROR_FOR_SWITCH
+
+        # No switch
+        self.assertFalse(hc.switchControlMode(
+            heading_error=mock_heading_error,
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+        self.assertEqual(hc.getControlModeID(), ControlModes.TACKABLE.value)
+
+    # Exit TACKABLE due to slow speed
+
+    def test_slowSpeed_exitTackable(self):
+        mock_speed = sailbot_constants.SPEED_THRESHOLD_FOR_JIBING_KNOTS + 0.1
+        mock_time = 0
+
+        # Enter TACKABLE from UNKNOWN
+        hc = HeadingController(mock_speed, mock_time, ControlModes.UNKNOWN.value)
+
+        # Make speed low and adjust time to meet the switch interval
+        mock_speed -= 0.2
+        mock_time += sailbot_constants.SWITCH_INTERVAL
+
+        self.assertTrue(hc.switchControlMode(
+            heading_error=0.5,
+            boat_speed=mock_speed,
+            current_time=mock_time
+        ))
+
+        # Should not be in TACKABLE or UNKNOWN
+        self.assertTrue(hc.getControlModeID() != ControlModes.TACKABLE.value)
+        self.assertTrue(hc.getControlModeID() != ControlModes.UNKNOWN.value)
+
+    # Exit TACKABLE due to reaching the heading
```

Comment:
similar to previous, don't think we need these type of comments. Can embed the information in test method name